### PR TITLE
refactor: add errorlint and refactor check error

### PR DESCRIPTION
**Pull request by hoanguyenkh**
## Summary by CodeRabbit

- **Refactor**
	- Improved error handling and message formatting across various components for better clarity and consistency in logs and output.

### CODE REVIEW

**Review by MSevey — DISMISSED** [review]

File: node/full_client.go [hunk 1]
```diff
@@ -871,7 +871,7 @@
 				c.Logger.Error("wanted to publish ResultEvent, but out channel is full:", full, "result:", result, "query:", result.Query)
 			}
 		case <-sub.Canceled():
-			if sub.Err() == cmpubsub.ErrUnsubscribed {
+			if errors.Is(sub.Err(), cmpubsub.ErrUnsubscribed) {
```
[the remark this answers is not in the document]
Comment (tuxcanfly) [reply node/full_client.go:874]: Good catch.

File: types/signed_header_test.go [hunk 2]
```diff
@@ -115,8 +116,11 @@ func testVerify(t *testing.T, trusted *SignedHeader, untrustedAdj *SignedHeader,
 				return
 			}
 
-			reason := err.(*header.VerifyError).Reason
-			expectedReason := test.err.(*header.VerifyError).Reason
+			var verifyError *header.VerifyError
+			errors.As(err, &verifyError)
```
Comment (tuxcanfly) [inline types/signed_header_test.go:120]:
```suggestion
			assert.True(t, errors.As(err, &verifyError))
```

File: state/txindex/kv/kv.go [hunk 3]
```diff
@@ -52,7 +53,7 @@ func (txi *TxIndex) Get(hash []byte) (*abci.TxResult, error) {
 
 	rawBytes, err := txi.store.Get(txi.ctx, ds.NewKey(hex.EncodeToString(hash)))
 	if err != nil {
-		if err == ds.ErrNotFound {
+		if errors.Is(err, ds.ErrNotFound) {
```
Comment (tuxcanfly) [inline state/txindex/kv/kv.go:56]: consider wrapping error below in line 68:

```
	if err != nil {
		return nil, fmt.Errorf("error reading TxResult: %v", err)
	}
```

**Review by tuxcanfly — CHANGES_REQUESTED** [review]
Minor nits for consistency.

File: state/txindex/kv/kv.go (hunk 3 above)
Comment (hoank101) [reply state/txindex/kv/kv.go:56]: thank you for feedback, i think still use this refactor, if change maybe break the previous issues -> https://github.com/rollkit/rollkit/pull/1585

File: node/full_client.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (hoank101) [reply node/full_client.go]: thank, i updated

**Review by tuxcanfly — APPROVED** [review]
OK.

**Review by tzdybal — APPROVED** [review]

**Pull request merged**